download/download.go

```diff
@@ -114,13 +116,24 @@ func (d *Downloader) loop() {
 	var retry int
 
 	for {
-		err := d.oneShot(ctx)
+		longPoll, err := d.oneShot(ctx)
 		var delay time.Duration
 
 		if err == nil {
-			min := float64(*d.config.Polling.MinDelaySeconds)
-			max := float64(*d.config.Polling.MaxDelaySeconds)
-			delay = time.Duration(((max - min) * rand.Float64()) + min)
+			if !longPoll {
+				if d.config.Polling.LongPollingTimeoutSeconds != nil {
+					d.config.Polling.LongPollingTimeoutSeconds = nil
+				}
+
+				// revert the response header timeout value on the http client's transport
+				if *d.client.Config().ResponseHeaderTimeoutSeconds == 0 {
+					d.client = d.client.SetResponseHeaderTimeout(&d.respHdrTimeoutSec)
+				}
+
+				min := float64(*d.config.Polling.MinDelaySeconds)
+				max := float64(*d.config.Polling.MaxDelaySeconds)
+				delay = time.Duration(((max - min) * rand.Float64()) + min)
```

Comment:
I think time.Duration expects nanoseconds...? So we'd need something like
```suggestion
				delay = time.Duration((((max - min) * rand.Float64()) + min)*time.Second)
```
...but I could be missing something 🤔

Reply:
I think we handle that [here](https://github.com/open-policy-agent/opa/blob/main/download/config.go#L48).

Reply:
Ah cool. It's a bit unfortunate to re-use the variable named `...Seconds` for that, though, I suppose. 🤔